### PR TITLE
[StructuralMechanicsApplication] Clean up the base load conditions

### PR DESCRIPTION
**Pull request by loumalouomega**
This add missing methods (clone, create, copy...) and clean ups the code (this make some warnings to appear some times)

### CODE REVIEW

File: applications/StructuralMechanicsApplication/custom_conditions/base_load_condition.cpp [hunk 1]
```diff
@@ -19,330 +20,401 @@
...
-            for(unsigned int k = 0; k<dim; ++k)
-            {
-                rValues[index + k] = Velocity[k];
-            }
+    if(dim == 2) {
+        for (SizeType i = 0; i < number_of_nodes; ++i) {
+            const SizeType index = i * block_size;
+            rResult[index    ] = GetGeometry()[i].GetDof(DISPLACEMENT_X,pos    ).EquationId();
+            rResult[index + 1] = GetGeometry()[i].GetDof(DISPLACEMENT_Y,pos + 1).EquationId();
+            if (this->HasRotDof())
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_conditions/base_load_condition.cpp:150]: I don't fully understand this check:
`virtual bool HasRotDof(){return (GetGeometry()[0].HasDofFor(ROTATION_X) && GetGeometry().size() == 2);};`

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_conditions/base_load_condition.cpp:150]: It was already there...

File: applications/StructuralMechanicsApplication/custom_conditions/base_load_condition.cpp [hunk 2]
```diff
@@ -19,330 +20,401 @@
...
     KRATOS_CATCH( "" )
-    }
+}
+/***********************************************************************************/
+/***********************************************************************************/
 
-} // Namespace Kratos
+void BaseLoadCondition::save( Serializer& rSerializer ) const
+{
+    KRATOS_SERIALIZE_SAVE_BASE_CLASS( rSerializer, BaseType )
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_conditions/base_load_condition.cpp:412]: I would have left this in the header ...
Dont know if it makes a difference, up to you

**Review by philbucher — APPROVED** [review]
A lot of things show up as diff, you did only cleaning, no refactoring right?

File: applications/StructuralMechanicsApplication/custom_conditions/base_load_condition.cpp (hunk 2 above)
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_conditions/base_load_condition.cpp:412]: Well, in this case does not make a difference.

This is just refactoring, and adding missing methods

**Pull request merged**